test/controllers/surveys_controller_test.rb

```diff
@@ -53,21 +79,35 @@ class SurveysControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "should show survey" do
-    get survey_url(@survey)
-    assert_response :success
+    @survey = create(:survey, user: @user, published: true)
+
+    sign_in @user
+    get survey_path(@survey)
+    assert_redirected_to survey_path(@survey)
   end
 
   test "should get edit" do
+    @survey = create(:survey, user: @user)
+
+    sign_in @user
     get edit_survey_url(@survey)
     assert_response :success
   end
 
   test "should update survey" do
-    patch survey_url(@survey), params: { survey: { name: @survey.name, published: @survey.published, user_id: @survey.user_id } }
-    assert_redirected_to survey_url(@survey)
+    @survey = create(:survey, user: @user)
+
+    sign_in @user
+    new_name = "My updated survey"
+    patch survey_path(@survey), params: { survey: { name: new_name, published: true } }
```

Comment:
Great work, I would just move the published to where you're creating the factory, there's no need to update it here, just the name will be enough

Reply:
Nice! I will change that.